jparest/src/test/java/uk/gov/homeoffice/digital/sas/jparest/controller/ResourceApiControllerTest.java

```diff
@@ -582,6 +583,309 @@ void update_idExistsOnRequestPathAndDoesNotExistOnBody_verifyPayloadIsValidatedW
 
     // endregion
 
+    // region batch update
+    @Test
+    @Transactional
+    void batchUpdate_resourcesExist_persistsChanges() throws JsonProcessingException {
+
+        String payloadOne = PayloadCreator.createPayload(Map.of(DESCRIPTION_FIELD_NAME, "Dummy Entity C One",
+            INDEX_FIELD_NAME, 1, TENANT_ID_FIELD_NAME, TENANT_ID));
+        String payloadTwo = PayloadCreator.createPayload(Map.of(DESCRIPTION_FIELD_NAME, "Dummy Entity C Two",
+            INDEX_FIELD_NAME, 2, TENANT_ID_FIELD_NAME, TENANT_ID));
+
+        //create new resources
+        var controller = getResourceApiController(DummyEntityC.class);
+        var apiResponseOne = controller.create(TENANT_ID, payloadOne);
+        assertThat(apiResponseOne.getItems()).hasSize(1);
+        var createdResource = apiResponseOne.getItems().get(0);
+
+        var apiResponseTwo = controller.create(TENANT_ID, payloadTwo);
+        assertThat(apiResponseTwo.getItems()).hasSize(1);
+        var createdResource2 = apiResponseTwo.getItems().get(0);
+
+        //get the newly created resources
+        var getResponseOne = controller.get(TENANT_ID, createdResource.getId());
+        var getResourceOne = getResponseOne.getItems().get(0);
+        assertThat(getResourceOne.getDescription()).isEqualTo("Dummy Entity C One");
+
+        var getResponseTwo = controller.get(TENANT_ID, createdResource2.getId());
+        var getResourceTwo = getResponseTwo.getItems().get(0);
+        assertThat(getResourceTwo.getDescription()).isEqualTo("Dummy Entity C Two");
+
+        //create update payload
+        String updatedResourceOnePayload = PayloadCreator.createPayload(Map.of(ID_FIELD_NAME, createdResource.getId(),
+            DESCRIPTION_FIELD_NAME, "Updated Dummy Entity C One",
+            INDEX_FIELD_NAME, 2, TENANT_ID_FIELD_NAME, TENANT_ID));
+
+        String updatedResourceTwoPayload = PayloadCreator.createPayload(Map.of(ID_FIELD_NAME, createdResource2.getId(),
+            DESCRIPTION_FIELD_NAME, "Updated Dummy Entity C Two",
+            INDEX_FIELD_NAME, 2, TENANT_ID_FIELD_NAME, TENANT_ID));
```

Comment:
nit: If i'v read correctly, the second created resource has an initial `index` value of 2, maybe this should be 3 here to assert a difference.

Reply:
updated